pkg/k8shandler/util.go

```diff
@@ -240,17 +224,7 @@ func (elasticsearchRequest *ElasticsearchRequest) isValidConf() error {
 		}
 	}
 
-	if !isValidRedundancyPolicy(dpl) {
-		if err := updateConditionWithRetry(dpl, v1.ConditionTrue, updateInvalidReplicationCondition, client); err != nil {
-			return err
-		}
-		return fmt.Errorf("Wrong RedundancyPolicy selected '%s'. Choose different RedundancyPolicy or add more nodes with data roles", dpl.Spec.RedundancyPolicy)
-	} else {
-		if err := updateConditionWithRetry(dpl, v1.ConditionFalse, updateInvalidReplicationCondition, client); err != nil {
-			return err
-		}
-	}
-
+	// TODO: replace this with a validating web hook to ensure field is immutable
```

Comment:
Can this compromise a CR's status conditions and in turn operator behavior by manual kubectl edits without solving this TODO?

Reply:
currently we validate if UUIDs have changed, a webhook to make the field immutable would be a better solution

Reply:
I understand? But can someone accidentally break the CR's UUIDs and break his cluster without the dropped logic?

Reply:
this comment has no direct impact on this pr, it is more a note to address it in a better way in the future

Reply:
I mean more the `isValidRedudancyPolicy` that is dropped. By dropping this check, can someone accidentally break the CR's UUIDs and in turn the cluster?

Reply:
the `isValidRedundancyPolicy` is already covered by crd validation [1] and has nothing to do with UUIDs. The removal of the check and addition of the comment are two independent things.

[1] https://github.com/openshift/elasticsearch-operator/blob/master/manifests/4.6/elasticsearches.crd.yaml#L28-L35